### DISCUSSION TITLE
OCILIB C API otext* Chinese Characters

### DISCUSSION THREAD

**Discussion by WinThuLatt** [vrogier/ocilib#267] [Q&A]
When OCI_GetString returns chinese characters, it returns L" inverted question marks. 

When I see the data in Oracle through Toad, I am able to see the characters. 

I have been trying to look up online on solutions. I have tried to change the code page. set the env variable manually before calling OCI_initialize.  

So far, none of them is working. 

Is it possible to get the data from NVARCHAR2 as bytes and reconstruct the chinese unicode string?

**Comment by vrogier** [comment]
Hi, 
Which ocilib charset version (ansi / wide) are you using?

**Reply by WinThuLatt** [reply]
I used the OCI_GetCharset and it returned 2. I suppose it is wide charset. 

Another thing I have tried is to call 
setlocale(LC_CTYPE, "UTF-8");
before OCI_Initialize

It didn't work as well.

**Reply by vrogier** [reply]
What is the type of your columns varchar or nvarchar?
What is the charset of the oracle client?
If you want to use utf8, you have to use the ansi (ociliba.dll) version of OCILIB and set the nls_lang environment variable to utf8

**Reply by WinThuLatt** [reply]
Hi, 
Columns are in NVARCHAR2

I used this to query the values. 
SELECT * FROM V$NLS_PARAMETERS
The NLS_CHARACTERSET is WE8MSWIN1252

select sys_context('USERENV', 'NLS_TERRITORY') from dual; 
return "AMERICA"

If you want to use utf8, you have to use the ansi (ociliba.dll) version of OCILIB  <- I am not sure how to achieve this. 
In the lib folder, I see both ociliba.dll and ocilibw.dll. Do you mean, I need to delete ocilibw.dll? 

Edit: If you want to use utf8, you have to use the ansi (ociliba.dll) version of OCILIB <- I updated the project to use ociliba.lib and recompiled. It didn't work as well
